plugins/builds/index.js

```diff
@@ -27,30 +27,30 @@ const workflowParser = require('screwdriver-workflow-parser');
  * @param  {Boolean}  [config.start]        Whether to start the build or not
  * @return {Promise}
  */
-function createBuild({ jobFactory, buildFactory, eventFactory, pipelineId, jobName, username,
-    scmContext, build, start }) {
-    return Promise.all([
-        eventFactory.get(build.eventId),
-        jobFactory.get({
-            name: jobName,
-            pipelineId
-        })
-    ]).then(([event, job]) => {
-        if (job.state === 'ENABLED') {
-            return buildFactory.create({
-                jobId: job.id,
-                sha: build.sha,
-                parentBuildId: build.id,
-                eventId: build.eventId,
-                username,
-                configPipelineSha: event.configPipelineSha,
-                scmContext,
-                start: start !== false
-            });
-        }
-
-        return null;
+async function createBuild({ jobFactory, buildFactory, eventFactory, pipelineId,
+    jobName, username, scmContext, build, start }) {
+    const event = await eventFactory.get(build.eventId);
+    const job = await jobFactory.get({
+        name: jobName,
+        pipelineId
     });
+    const prRef = event.pr.ref ? event.pr.ref : '';
```

Comment:
is `pr` guaranteed to be an object here ? else we should do a check on `event.pr` as well.

Reply:
@jithin1987 
yes, `pr` is guaranteed to ben an object here. Please see this link.
https://github.com/screwdriver-cd/models/blob/master/lib/eventFactory.js#L416